nossomercadolivre-api/src/main/java/com/github/nossomercadolivre/User.java

```diff
@@ -1,36 +1,43 @@
 package com.github.nossomercadolivre;
 
-import com.github.nossomercadolivre.exception.InvalidPasswordException;
-import org.springframework.data.annotation.CreatedDate;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import static javax.persistence.GenerationType.IDENTITY;
 
-import javax.persistence.*;
-import javax.validation.constraints.*;
-import java.io.Serializable;
 import java.time.Instant;
 import java.util.Objects;
 
-import static javax.persistence.GenerationType.IDENTITY;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EntityListeners;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.validation.constraints.Email;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 
 @Entity
-@Table
-public class User implements Serializable {
+@EntityListeners(AuditingEntityListener.class)
```

Comment:
Pq temos essa audioria aqui?

Reply:
Se não muito me engano, para a @CreatedDate funcionar, precisa da @EntityListeners(AuditingEntityListener.class)

Talvez haja outras formas mais limpas de fazer a @CreatedDate funcionar!

Reply:
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Aqui, para mim é assim: se tem pronto no runtime eu prefiro usar :). Você só precisa mandar um "new" no instante e está resolvido. Este é um codigo que todo mundo deveria conhecer, já que faz parte da linguagem. Já o @CreatedDate assume que a pessoa precisa manjar da annotation e entender que, de algum jeito, o hibernate(implementaçao da jpa) vai colocar uma data. Isso aumenta a carga intrínseca(dificuldade de entendimento). Sem contar que faz o objeto ser criado em estado inválido, já que nasce sem instante de criaçao.

Reply:
Concordo com: "o objeto ser criado em estado inválido", e também acho isso ruim.
 ✅Alterei